docs/efile-submission-runbook.md

```diff
@@ -0,0 +1,43 @@
+# EfileSubmission runbook
+
+Date: 2021-08-02
+
+Authors: Asheesh Laroia, Travis Grathwell
+
+## Purpose
+Engineers might have to handle EfileSubmission records during on-call.
+This [runbook](https://en.wikipedia.org/wiki/Runbook) is a guide on how to do that.
+
+## Typical operation
+
+An EfileSubmission record is created during intake when clients have finished submitting the information needed to submit an Advance CTC return.
+
+The EfileSubmission has a state machine attached (EfileSubmissionStateMachine) to it. The standard flow is:
+
+- A record is created as `:new`.
+- When it is transitioned to `:preparing`, we create a BundleSubmissionJob and update the tax return status for Hub users. BundleSubmissionJob validates the client's address, generates a 1040 PDF for the client and Hub users to see, and creates a submission ZIP file. It transitions the EfileSubmission to `:queued`.
+- When it is transitioned to `:queued`, we create a SendSubmissionJob. This launches gyr-efiler to send the submission ZIP file to the IRS. This sets the state to `:transmitted`.
+- Over time, a cron job will run that polls the IRS for status of transmitted submissions. It launches gyr-efiler with the IRS submission IDs. When the IRS responds with the status, the cron job marks the EfileSubmission as `:accepted` or `:rejected` and updates the TaxReturn status.
+
+An EfileSubmission can transition to `:resubmitted` if it submission ZIP never reached the IRS.
```

Comment:
When clicking the "resubmit" button, the current EfileSubmission object will always transition to resubmitted. In the case that it was never transmitted to the IRS, we transition from resubmitted back to preparing (re-prepare the same submission). In the case that we have already transmitted the submission to the IRS, that IRS ID cannot be used again so we create a new EfileSubmission object. Both of these actions happen in the after_transition callback to :resubmitted.